### PR TITLE
refactor(tests): change the text result check to goldenfile

### LINKED ISSUE #5001: Consider change the text result check to goldenfile
**Summary**

There are many `assert_blocks_sorted_eq` in databend rust code, to make life easier, let's change to goldenfile.
goldenfile example:
https://github.com/datafuselabs/databend/blob/f25c34917b415ab42308a99637b5513aa4ea76df/common/ast/tests/it/rule.rs#L120-L137

### PR DESCRIPTION
**Pull request by PsiACE**
… demo

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- [x] a basic demo -  `cargo test --package databend-query --test goldenfiles`
- [ ] change all text result checks to goldenfile
- [ ] merge to tests/it

## Changelog

- Improvement
- Build/Testing/CI

## Related Issues

Fixes #5001

### CODE REVIEW

**Comment by BohuTANG** [comment]
Let's make core team life easier!

File: query/tests/goldenfiles/interpreters/interpreter_show_tables.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        {
+            let plan = PlanParser::parse(ctx.clone(), "create table bend(a Int)").await?;
+            let executor = InterpreterFactory::get(ctx.clone(), plan.clone())?;
+            let _ = executor.execute(None).await?;
+        }
+    }
+
+    // show tables.
+    {
+        let mut mint = Mint::new("tests/goldenfiles/data");
```
Comment (sundy-li) [inline query/tests/goldenfiles/interpreters/interpreter_show_tables.rs:61]: Can we use a macro to wrap these codes?

Comment (PsiACE) [reply query/tests/goldenfiles/interpreters/interpreter_show_tables.rs:61]: Yes, a function or a macro. The first commit is just for a simple demo.

**Comment by PsiACE** [comment]
I should have tried fewer changes, small PRs. We can easily replace unsorted/sorted assert, but I'm not sure how to handle regulars yet.

**Comment by sundy-li** [comment]
> I'm not sure how to handle regex yet

1. Avoid dynamic results in tests
2. Use SQL function to replace the dynamic result ( replace, regexp_xxx)

**Comment by PsiACE** [comment]
Let me close it and do some minor refactorings first.

**Pull request closed without merging**